includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1213,4 +1174,26 @@ protected function build_display_items( $itemized_display_items = false ) {
 			],
 		];
 	}
+
+	/**
+	 * Calculates whether Apple Pay is enabled for this store.
+	 * The option value is not stored in the database, and is calculated
+	 * using this function instead, and the values is returned by using the pre_option filter.
+	 *
+	 * The option value is retrieved for inbox notifications.
+	 *
+	 * @param mixed $value The value of the option.
+	 */
+	public function get_option_is_apple_pay_enabled( $value ) {
+		// Return a random value (1 or 2) if the account is live and payment request buttons are enabled.
+		if (
+			WC_Payments::get_gateway()->is_enabled()
```

Comment:
Based on feedback from my PR I've included the gateway instance in a `$this->gateway` private variable now, so it would make sense to replace all instances of `WC_Payments::get_gateway()` with `$this->gateway`.

Reply:
Sounds good 👍